recbole/model/general_recommender/multidae.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# @Time   : 2020/12/14
+# @Author : Yihong Guo
+# @Email  : gyihong@hotmail.com
+
+r"""
+MultiDAE
+################################################
+Reference:
+    Dawen Liang et al. "Variational Autoencoders for Collaborative Filtering." in WWW 2018.
+
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import numpy as np
+
+from recbole.utils import InputType
+from recbole.model.abstract_recommender import GeneralRecommender
+from recbole.model.init import xavier_normal_initialization
+
+
+class MultiDAE(GeneralRecommender):
+    r"""MultiDAE is a item-based model collaborative filtering model that simultaneously rank all items for user .
+
+    We implement the the MultiDAE model with only user dataloader.
+    """
+    input_type = InputType.PAIRWISE
+
+    def __init__(self, config, dataset):
+        super(MultiDAE, self).__init__(config, dataset)
+
+        layers = config["mlp_hidden_size"]
+        lat_dim = config['latent_dimendion']
+        self.drop_out = config['dropout_prob']
+
+        self.lat_dim = lat_dim
+        self.interaction_matrix = dataset.inter_matrix(form='csr').astype(np.float32)
+
+        self.encode_layer_dims = [self.n_items] + layers + [lat_dim]
+        self.decode_layer_dims = [self.lat_dim] + self.encode_layer_dims[::-1][1:]
+
+        self.encoder = self.mlp_layars(self.encode_layer_dims)
+        self.decoder = self.mlp_layars(self.decode_layer_dims)
+
+        # parameters initialization
+        self.apply(xavier_normal_initialization)
+
+    def get_rating_matrix(self, user_id):
+        return torch.tensor(self.interaction_matrix[user_id.cpu(), :].toarray(), device=self.device)
```

Comment:
```suggestion
        return torch.tensor(self.interaction_matrix[user_id.cpu(), :].toarray(), device=self.device)
```
I sugguest not to get a batch of user's interaction vector in this way. There is a more efficient way that you could refer to the implement of   DMF. 
https://github.com/RUCAIBox/RecBole/blob/511cda89deca404808bea1df2f5e98e17e9015f1/recbole/model/general_recommender/dmf.py#L141-L158